File: 02-Array/LeetCode-682.py
# 682. Baseball Game
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        for S in operations:
            if S == "C":
                stack.pop()
            elif S == "D":
                stack.append(stack[-1]*2)
            elif S == "+":
                stack.append(stack[-1]+stack[-2])
            else:
                stack.append(int(S))
        return sum(stack)

File: 03-Stack/LeetCode-682.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> record;
        for (string S : operations)
        {
            if (S == "C")
                record.pop();
            else if (S == "D")
                record.push(record.top()*2);
            else if (S == "+")
            {
                int f1 = record.top();
                record.pop();
                int f2 = record.top();
                record.push(f1);
                record.push(f1+f2);
            }
            else
                record.push(stoi(S));
        }
        int ans = 0;
        while (!record.empty())
        {
            ans += record.top();
            record.pop();
        }
        return ans;
    }
};
